File: msgraph-mail/python/models/extension.py
from __future__ import annotations
from kiota_abstractions.serialization import Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

from . import entity

class Extension(entity.Entity):
    def __init__(self,) -> None:
        """
        Instantiates a new Extension and sets the default values.
        """
        super().__init__()

    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> Extension:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: Extension
        """
        if not parse_node:
            raise Exception("parse_node cannot be undefined")
        return Extension()

    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
        }
        super_fields = super().get_field_deserializers()
        fields.update(super_fields)
        return fields

    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if not writer:
            raise Exception("writer cannot be undefined")
        super().serialize(writer)

File: msgraph-mail/typescript/src/models/extension.ts
import { deserializeIntoEntity, serializeEntity, type Entity } from './entity';
import { type Parsable, type ParseNode, type SerializationWriter } from '@microsoft/kiota-abstractions';

export function createExtensionFromDiscriminatorValue(parseNode: ParseNode | undefined) {
    if(!parseNode) throw new Error("parseNode cannot be undefined");
    return deserializeIntoExtension;
}
export function deserializeIntoExtension(extension: Extension | undefined = {} as Extension) : Record<string, (node: ParseNode) => void> {
    return {
        ...deserializeIntoEntity(extension),
    }
}
export interface Extension extends Entity, Parsable {
}
export function serializeExtension(writer: SerializationWriter, extension: Extension | undefined = {} as Extension) : void {
        serializeEntity(writer, extension)
}
